SPECS/containerd2/containerd2.spec

```diff
@@ -0,0 +1,229 @@
+%global debug_package %{nil}
+%define upstream_name containerd
+%define commit_hash 207ad711eabd375a01713109a8a197d197ff6542
+
+Summary: Industry-standard container runtime
+Name: %{upstream_name}2
+Version: 2.0.0
+Release: 1%{?dist}
+License: ASL 2.0
+Group: Tools/Container
+URL: https://www.containerd.io
+Vendor: Microsoft Corporation
+Distribution: Azure Linux
+
+Source0: https://github.com/containerd/containerd/archive/v%{version}.tar.gz#/%{upstream_name}-%{version}.tar.gz
+Source1: containerd.service
+Source2: containerd.toml
+
+%{?systemd_requires}
+
+BuildRequires: git
+BuildRequires: golang < 1.23
```

Comment:
Is there a reason we need to pin to a lower version of golang?

Reply:
Great point. containerd 2.0 requires golang1.22 or above ([doc](https://github.com/containerd/containerd/blob/v2.0.0/BUILDING.md?plain=1#L28)) and runc is also built with both golang 1.22 and1.23 in its [CI workflow](https://github.com/opencontainers/runc/blob/v1.2.1/.github/workflows/test.yml#L27). so I think we can remove this golang version restriction. @mfrw  could you please confirm we could remove it?

Reply:
Confirmed by Falak and removed the golang version restriction.

Reply:
Ack on removal on golang version constraint.